File: perception-core/src/main/java/com/hh/uiperception/core/semantic/SemanticRole.java
package com.hh.uiperception.core.semantic;

/**
 * Android native semantic snapshot 第一版 role 集合。
 */
public enum SemanticRole {
    SCREEN("screen"),
    TOOLBAR("toolbar"),
    TABBAR("tabbar"),
    TAB("tab"),
    LIST("list"),
    GRID("grid"),
    LIST_ITEM("listitem"),
    SCROLL("scroll"),
    CARD("card"),
    SECTION("section"),
    DIALOG("dialog"),
    SHEET("sheet"),
    TEXT("text"),
    IMAGE("image"),
    BUTTON("button"),
    INPUT("input"),
    CHECKBOX("checkbox"),
    RADIO("radio"),
    SWITCH("switch"),
    SLIDER("slider"),
    PICKER("picker"),
    PROGRESS("progress"),
    GENERIC("generic"),
    WEBVIEW("webview"),
    LINK("link"),
    HEADING("heading"),
    NAVIGATION("navigation"),
    TEXTBOX("textbox"),
    SEARCHBOX("searchbox"),
    SPINBUTTON("spinbutton"),
    COMBOBOX("combobox"),
    LISTBOX("listbox"),
    TABLE_ROLE("table"),
    ROW("row"),
    CELL("cell"),
    COLUMNHEADER("columnheader"),
    ROWHEADER("rowheader"),
    FORM("form"),
    ARTICLE("article"),
    COMPLEMENTARY("complementary"),
    BLOCKQUOTE("blockquote"),
    CAPTION("caption"),
    GROUP("group"),
    TERM("term"),
    DEFINITION("definition"),
    SEPARATOR("separator"),
    METER("meter"),
    OPTION_ROLE("option"),
    STATUS("status"),
    PARAGRAPH("paragraph"),
    ROWGROUP("rowgroup");

    private final String snapshotName;

    SemanticRole(String snapshotName) {
        this.snapshotName = snapshotName;
    }

    public String snapshotName() {
        return snapshotName;
    }
}

File: ios/UIPerception/Capture/SemanticRole.swift
import Foundation

/// 语义角色枚举，对应 Android `SemanticRole.java`。
/// snapshotName 是 YAML 输出和 JSON wire 格式中使用的字符串。
enum SemanticRole: String {
    // native + web 共享
    case screen
    case toolbar
    case tabbar
    case tab
    case list
    case grid
    case listItem = "listitem"
    case scroll
    case card
    case section
    case dialog
    case sheet
    case text
    case image
    case button
    case input
    case checkbox
    case radio
    case switchRole = "switch"
    case slider
    case picker
    case progress
    case generic
    case webview
    case link
    case heading
    case navigation

    // web 专用（HTML ARIA），native 不产出
    case textbox
    case searchbox
    case spinbutton
    case combobox
    case listbox
    case tableRole = "table"
    case row
    case cell
    case columnheader
    case rowheader
    case form
    case article
    case complementary
    case blockquote
    case caption
    case group
    case term
    case definition
    case separator
    case meter
    case optionRole = "option"
    case status
    case paragraph
    case rowgroup

    var snapshotName: String { rawValue }
}
